### DISCUSSION TITLE
beamnrc and dosxyznrc job failure

### DISCUSSION THREAD

**Discussion by Kelvin-Ngowi** [nrc-cnrc/EGSnrc#1030] [Questions]
I recently installed my egsnrc and compilation and execution working correctly for EX16 mvp but not for other examples using a **Gui.** 
any help from you I would really appreciate it.

**Comment by rtownson** [comment]
What's the error that you're seeing? What examples are you trying? Please provide as many details, screenshots, etc. as you can for a more efficient response.

**Reply by Kelvin-Ngowi** [reply]
i can't see anything. like a warning, error etc.
i am excecuting 16MVp H20 PHANTOM beam source example using DOSXYZnrc GUI.
THE only feedback that i am getting is the job failure message.
i have tried to save each time i perform compilation as recommended but nothing changes.
can you please help

**Reply by rtownson** [reply]
For source 9 (shared library source) in DOSXYZnrc, you need to compile the accelerator as a shared library. This has to be done on the command line. Navigate to the accelerator directory (egs_home/BEAM_EX16MVp) and type `mingw32-make library`. Then, I recommend testing the accelerator with the input file you're using (if you haven't yet, you'll also need to type `mingw32-make`). E.g.

```
BEAM_EX16MVp -i EX16MVp -p 700icru
```
If that works, then you have some idea it should work in DOSXYZnrc as well (after it is compiled as a shared library).

For better error messages, run on the command line and/or check the .egslog file.

**Reply by Kelvin-Ngowi** [reply]
$ ming32-make library
bash: ming32-make: command not found

this is the message that i am getting suppose if follow your steps.
i was thinking of use a parallel beam from the front (SOURCE 0) as a source.
would i get different results if i try to use two different beam sources (say source 0 and source 9) one at a time?

**Reply by Kelvin-Ngowi** [reply]
one thing, there isn't any problem with beamnrc both in Gui and running use a cmd.
the only problem is A JOB FAILURE pop up i get if i want to execute via GUI.

**Reply by Kelvin-Ngowi** [reply]
things don't work in expected way not even if i change the beam source for DOSXYZnrc.
 please help me.

**Reply by Kelvin-Ngowi** [reply]
IT could be help full if you exactly how  does BEAM_EX16MVp -i EX16MVp -p 700icru command appear when it comes to dosxyznrc

**Reply by Kelvin-Ngowi** [reply]
$ dosxyznrc -i 16MVp_h2o_phantom_beamsource_example -p 700icru

***************** Error:
 Input file D:\EGSnrc\egs_home\dosxyznrc\16MVp_h2o_phantom_beamsource_example.egsinp does not exist.

***************** Quiting now.
 thats what i get if i try to excecute the programm from the command prompt

**Reply by rtownson** [reply]
> $ ming32-make library bash: ming32-make: command not found

There was a typo in my comment, it should be `mingw32-make library`. 

> would i get different results if i try to use two different beam sources (say source 0 and source 9) one at a time?

Unless source 0 and source 9 model identical sources, yes you will get different results.

> IT could be help full if you exactly how does BEAM_EX16MVp -i EX16MVp -p 700icru command appear

In your dosxyznrc input file, for source 9 you input these parameters. In the input file, it's a line that looks like this:
`BEAM_EX16MVp,EX16MVp,700icru`

> Input file D:\EGSnrc\egs_home\dosxyznrc\16MVp_h2o_phantom_beamsource_example.egsinp does not exist.

Does it really exist in that directory? It must not.

**Comment by Kelvin-Ngowi** [comment]
hi there 
i  have certainly recieved the following errors as i was trying to execute the mingw32-make library command via the cmd in a ex beam 16 mvp directory
''$ mingw32-make library
gfortran -shared -Wl,-Bsymbolic                                  -fPIC -O2 -mtune=native -DCOMPILE_TIME="\"Tue 07/18/2023 12:10 AM\"" -DGIT_HASH="\"unknown\"" -o D:\EGSnrc\egs_home\bin\win3264\BEAM_EX16MVp.dll libBEAM_EX16MVp_win3264.obj D:\EGSnrc\HEN_HOUSE\lib\win3264\egs_c_utils.obj D:\EGSnrc\HEN_HOUSE\lib\win3264\load_beamlib.obj  -LD:\EGSnrc\HEN_HOUSE\egs++\dso\win3264   -liaea_phsp
gfortran.exe: error: D:EGSnrcHEN_HOUSElibwin3264egs_c_utils.obj: No such file or directory
gfortran.exe: error: D:EGSnrcHEN_HOUSElibwin3264load_beamlib.obj: No such file or directory
mingw32-make: *** [D:\EGSnrc\HEN_HOUSE\makefiles\beam_makefile:120: D:\EGSnrc\egs_home\bin\win3264\BEAM_EX16MVp.dll] Error 1
''
what should I do?

**Reply by rtownson** [reply]
Does `mingw32-make` work? It looks like you haven't installed using `git clone`, which is the preferred option. Can you post your installation log, found in `HEN_HOUSE/log`?

**Reply by Kelvin-Ngowi** [reply]
[Uploading config_win3264.log…]()
there you go

**Reply by Kelvin-Ngowi** [reply]
i perfomed the installation using the git clone as reccomended and thats is why i can even compile and execute beam 16mvp example.

**Reply by Kelvin-Ngowi** [reply]
i perfomed the installation using the git clone as reccomended and thats is why i can even compile and execute beam 16mvp example.

**Reply by rtownson** [reply]
What happens if you type 'git status' while you are in that directory? I'm just seeing that it reports `-DGIT_HASH=""unknown""`. If you can compile the beam accelerator I don't understand why this command would fail. Do you have a different installation and/or the environment variables are inconsistent between them?

> [Uploading config_win3264.log…](https://github.com/nrc-cnrc/EGSnrc/discussions/1030)

This link isn't to a file.

**Reply by Kelvin-Ngowi** [reply]
[Uploading config_win3264.log…]()

**Reply by rtownson** [reply]
Still didn't upload the file. Also, did you check git as I suggested? I recommend starting fresh with a new git clone, checking out the branch from the pull request I linked, and trying again.

**Reply by Kelvin-Ngowi** [reply]
what exactly do you always check and correct in a log file  may be i can gain a knowledge and help others too.

**Reply by rtownson** [reply]
The log file gives a wide variety of clues. It reveals the version of mingw used, whether path lengths were too long or had spaces, where the code fails to compile, etc. It's much more efficient than asking questions.

Btw, yours still didn't work, and I'm not sure if you've followed all my suggestions.

**Comment by Kelvin-Ngowi** [comment]
i still dont know how to over come a job failure in dosxyznrc using a gui
it seems like there isnt an input file in dosxyznrc  so it can be executed.
where exactly does 16MVp_h2o_phantom_beamsource_example input file exist

**Reply by rtownson** [reply]
You can copy it out of the `egs_home/dosxyznrc/examples` directory.

The GUI just lets you know the job didn't complete normally, but you still need to look at the output from the simulation to see why.

**Reply by Kelvin-Ngowi** [reply]
i tried this earlier and it worked out perfectly but right now i cant no more

**Reply by rtownson** [reply]
I don't understand what doesn't work, can you explain in more detail.

**Reply by Kelvin-Ngowi** [reply]
Particles to be simulated: photon only


 Medium                AE        AP
 H2O700ICRU          0.700     0.010

 Will perform charged-particle range rejection against voxel boundaries
 if particle energy is below      3.00000 MeV.

 Photons will be split   40 times



 ***************************************************************

  Histories to be simulated for this run         1000

  Histories to be analyzed after this run        1000

 ***************************************************************


                                       NP   ENERGY  Q REGION    X       Y       Z      U      V      W      LATCH  WEIGHT

   Elapsed wall clock time to this point=       0.145 s


   CPU time so far for this run =       0.000 s


 BATCH #  TIME-ELAPSED  TOTAL CPUTIME  RATIO  TIME OF DAY  RNG pointers

     1          0.0            0.0      0.00    21:31:13   ixx jxx =   97  33
**You have to load and initialize the library first**

**Reply by rtownson** [reply]
It's just complaining that the BEAMnrc accelerator isn't compiled as a shared library, as expected.

**Reply by Kelvin-Ngowi** [reply]
caesa@calvingaming MINGW64 /D/EGSnrc/egs_home/BEAM_EX16MVp (fix-win-compile)
$ mingw32-make library
gfortran -shared -Wl,-Bsymbolic                                  -fPIC -O2 -mtune=native -DCOMPILE_TIME="\"Thu 07/20/2023 08:47 PM\"" -DGIT_HASH="\"unknown\"" -o D:\EGSnrc\egs_home\bin\win3264\BEAM_EX16MVp.dll libBEAM_EX16MVp_win3264.obj D:\EGSnrc\HEN_HOUSE\lib\win3264\egs_c_utils.obj D:\EGSnrc\HEN_HOUSE\lib\win3264\load_beamlib.obj  -LD:\EGSnrc\HEN_HOUSE\egs++\dso\win3264   -liaea_phsp
gfortran.exe: error: D:EGSnrcHEN_HOUSElibwin3264egs_c_utils.obj: No such file or directory
gfortran.exe: error: D:EGSnrcHEN_HOUSElibwin3264load_beamlib.obj: No such file or directory
mingw32-make: *** [D:\EGSnrc\HEN_HOUSE\makefiles\beam_makefile:120: D:\EGSnrc\egs_home\bin\win3264\BEAM_EX16MVp.dll] Error 1

**Comment by Kelvin-Ngowi** [comment]
a 2023a version of egsnrc somes to have alot of errors sir.
shall i shift to the 2021 and below version as a way to escape from all these errors?

**Reply by rtownson** [reply]
I believe all the issues are fixed by https://github.com/nrc-cnrc/EGSnrc/pull/972, so I suggest trying that. You're also welcome to try v2021. As always, I recommend installing using git not using the zip file.

**Reply by Kelvin-Ngowi** [reply]
does the previous versions have the same cases?

**Reply by Kelvin-Ngowi** [reply]
does the previous versions have the same cases?

**Comment by rtownson** [comment]
> does the previous versions have the same cases?

Do you mean examples? Yes, see the release notes here: https://github.com/nrc-cnrc/EGSnrc/wiki/Release-notes

**Reply by Kelvin-Ngowi** [reply]
the other thing is i cant find a 3ddose file

**Reply by Kelvin-Ngowi** [reply]
sorry sir, correct me if i am wrong about the following
1. i have to create build and compile the accelerator
2.  i have to create water phantom and select an input beam from the available ones 
3. i have to perform simulation and then a 3ddose file is generated as an output
4. final i have to use a 3ddose file together with other programs (say victoria) and study the pdds and dose profiles
5. procedure 3 has to be done as many times as it can be done by changing the energy and beam radius until the right one is obtained (when modelling a linac head of a certain nominal energy)

**Reply by rtownson** [reply]
Generally, yes. Though modelling a linac head may be considerably more complex that just adjusting beam energy and radius.

**Reply by Kelvin-Ngowi** [reply]
i expected to have a 3ddose file as an output , but unfortunately i cant find it. i am using window 11 OS

**Reply by rtownson** [reply]
Solve the installation issues first. If anything is failing to compile, that's a problem. There are also workarounds like virtualbox.
